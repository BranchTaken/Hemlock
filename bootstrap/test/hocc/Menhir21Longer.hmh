# Implementation of `longer.mly` grammar from:
#
#   https://gitlab.inria.fr/fpottier/menhir/-/issues/21

hocc
    token INFIXOP0
    token EQUAL "="
    token NAME
    token NUMERAL
    token LET "let"
    token REC "rec"
    token IN "in"
    token AND "and"
    token EOF

    start Commandline ::=
      | "let" NAME Term EOF
      | Term EOF

    nonterm Term ::=
      | InfixTerm_
      | "let" NAME "=" InfixTerm "in" Term
      | "let" "rec" Fs "in" Term

    nonterm InfixTerm ::=
      | NUMERAL

    nonterm InfixTerm_ ::=
      | NUMERAL
      | InfixTerm INFIXOP0 InfixTerm

    # Menhir's `separated_nonempty_list` is expanded here to `Fs` and `FsTl`.
    nonterm Fs ::=
      | RecursiveClause FsTl

    nonterm FsTl ::=
      | "and" RecursiveClause FsTl
      | epsilon

    nonterm RecursiveClause ::=
      | NAME "=" Term
