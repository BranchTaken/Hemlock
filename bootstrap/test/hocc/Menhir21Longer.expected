hocc: Parsing "./Menhir21Longer.hmh"
hocc: Generating PGM(1) specification
hocc: 0 precedences, 11 tokens, 8 non-terminals (1 start), 13 productions
hocc: LR(1) item set compatibility: weak
hocc: Generating LR(1) item set closures (+^.=add/split/merge)+++++++++++++++++++.++++++++..+............++++
hocc: Generating 33 LR(1) states
hocc: 0 unresolvable conflicts in 0 states
hocc: 0 unreachable states
hocc: 0 remergeable states
hocc: Searching for unused precedences/tokens/non-terminals/productions
hocc: Generating text report
hocc: Generating hocc report
hocc: Writing "./hocc/Menhir21Longer.txt"
hocc: Writing "./hocc/Menhir21Longer.hmh"
